src/utl/include/utl/Logger.h

```diff
@@ -174,11 +176,22 @@ class Logger
   // Note: these methods do no escaping so avoid special characters.
   template <typename T,
             typename U = std::enable_if_t<std::is_arithmetic<T>::value>>
-  inline void metric(const std::string_view metric, T value)
+  inline void metric(const std::string_view metric_name, T value)
   {
-    std::ostringstream oss;
-    oss << std::defaultfloat << std::setprecision(6) << value;
-    log_metric(std::string(metric), oss.str());
+    const std::string name = std::string(metric_name);
+    if (std::isinf(value)) {
+      if (value < 0) {
+        metric(name, "-inf");
+      } else {
+        metric(name, "inf");
+      }
+    } else if (std::isnan(value)) {
+      metric(name, "nan");
```

Comment:
In the [python json module](https://docs.python.org/3/library/json.html):
```
 If allow_nan is true, their JavaScript equivalents (NaN, Infinity, -Infinity) will be used.
```
Similarly with parse_constant in json.load.  We should match their naming.

Reply:
Will do